Packs/TrendMicroEmailSecurity/Integrations/TrendMicroEmailSecurityEventCollector/TrendMicroEmailSecurityEventCollector_test.py

```diff
@@ -0,0 +1,356 @@
+    """
+    mocker.patch.object(mock_client, "get_logs_request", side_effect=event_mock)
+    _, next_token = fetch_by_event_type(mock_client, "", "", limit, None, "", False)
+
+    assert not next_token
+
+
+@pytest.mark.parametrize(
+    "event_mock, limit, expected_results",
+    [
+        (({},), 1, {"len_events": 0, "call_count": 1}),
+        (
+            ({"nextToken": "test", "logs": [{}]}, {}),
+            2,
+            {"len_events": 1, "call_count": 2},
+        ),
+        (({"logs": [{}]},), 1, {"len_events": 1, "call_count": 1}),
+        (
+            ({"nextToken": "test", "logs": [{}]}, {"logs": [{}]}),
+            2,
+            {"len_events": 2, "call_count": 2},
+        ),
+        ((NoContentException(),), 1, {"len_events": 0, "call_count": 1}),
+        (
+            ({"nextToken": "test", "logs": [{}]}, NoContentException()),
+            2,
+            {"len_events": 1, "call_count": 2},
+        ),
+        (
+            ({"nextToken": "test", "logs": [{}]}, {"nextToken": "test", "logs": [{}]}),
+            2,
+            {"len_events": 2, "call_count": 2, "next_token": "test"},
+        ),
+    ],
+)
+def test_fetch_by_event_type(
+    mocker,
+    mock_client: Client,
+    event_mock: tuple[dict],
+    limit: int,
+    expected_results: dict,
+):
+    """
+    Given:
+        - args for fetch_by_event_type function
+    When:
+        - run fetch_by_event_type function and return from api request response by event mock
+    Then:
+        - Ensure that when no events are returned from the API
+          it exits the while loop and returns 0 events.
+        - Ensure that when in the second iteration no events are returned from the API,
+          it exits the while loop and returns 1 event and the api call is called twice.
+        - Ensure that when no `nextToken` returns from the API
+          it exits the while loop and returns 1 event
+        - Ensure that when in the second iteration `nextToken` does not return from the API
+          it exits the while loop and returns 1 event and the api call is called twice.
+        - Ensure that when `NoContentException` returns from the API
+          it exits the while loop and returns 0 events
+        - Ensure that when in the second iteration `NoContentException` returns from the API
+          it exits the while loop and returns 1 event and the api call is called twice.
+        - Ensure that when the number of events returned from the API is equal to the `limit`
+          it exits the while loop and returns the returned events and the `nextToken`.
+    """
+    mock_api = mocker.patch.object(
+        mock_client, "get_logs_request", side_effect=event_mock
+    )
+    events, next_token = fetch_by_event_type(mock_client, "", "", limit, None, "", False)
+
+    assert len(events) == expected_results["len_events"]
+    assert mock_api.call_count == expected_results["call_count"]
+    assert next_token == expected_results.get("next_token")
```

Comment:
`.get` on purpose? 🤔

Reply:
It is not always in expected_results
Part of the UT simulates that no next_token is returned at all